pkg/pillar/types/zedmanagertypes.go

```diff
@@ -132,6 +133,36 @@ type AppInstanceStatus struct {
 	ErrorAndTimeWithSource
 }
 
+func (status AppInstanceStatus) LogCreate() {
+	logObject := base.NewLogObject(base.AppInstanceLogType, status.DisplayName,
+		status.UUIDandVersion.UUID, status.Key())
+	if logObject == nil {
+		return
+	}
+	logObject.CloneAndAddField("state", status.State).AddField("restart-in-progress", status.RestartInprogress).
+		AddField("purge-in-progress", status.PurgeInprogress).Infof("App instance status")
+}
+
+func (status AppInstanceStatus) LogModify(old interface{}) {
+	logObject := base.LookupLogObject(status.Key())
+	if logObject == nil {
+		logObject = base.NewLogObject(base.AppInstanceLogType, status.DisplayName,
+			status.UUIDandVersion.UUID, status.Key())
+		if logObject == nil {
+			return
+		}
+	}
+
+	// XXX Not comparing/using old app instance status for the demo
```

Comment:
I'd add some code to compare the three fields you log with the old, just to make this an example of the indended use pattern.

Reply:
Also, should check HasError and do a log.Error if it is set (or if it has changed from old)

Reply:
will do